conman/routes/models.py

```diff
@@ -74,6 +84,64 @@ def handle(self, request, path):
         # Deal with the request
         return handler.handle(request, path)
 
+    def move_to(self, new_url, *, move_children):
+        """
+        Move this Route to a new url.
+
+        Requires `move_children` as a keyword argument. If True, all child
+        Routes will be moved along with this parent. Otherwise, when False, any
+        and all children will remain where they are.
+
+        In the event of a clash, an IntegrityError will be raised.
+        """
+        if move_children:
+            Route.objects.move_branch(self.url, new_url)
+            self.url = new_url
+        else:
+            old_url, self.url = self.url, new_url
+            try:
+                self.save()
+            except IntegrityError:
+                self.url = old_url
+                raise
```

Comment:
I think I might be too tired to make a sensible decision on this at the moment, despite having had a change of mind about this. Perhaps it'd be a good one to discuss in person.